consensus/src/ordered_broadcast/types.rs

```diff
@@ -0,0 +1,780 @@
+    fn payload(chunk: &Chunk<P, D>, epoch: &Epoch) -> Vec<u8> {
+        let mut message = Vec::with_capacity(Chunk::<P, D>::SIZE + Epoch::SIZE);
+        chunk.write(&mut message);
+        epoch.write(&mut message);
+        message
+    }
+
+    /// Verify the Ack.
+    ///
+    /// This ensures that the partial signature is valid for the given chunk and epoch,
+    /// using the provided identity (which contains the BLS public polynomial).
+    ///
+    /// Returns true if the signature is valid, false otherwise.
+    pub fn verify(&self, namespace: &[u8], identity: &poly::Public) -> bool {
+        // Construct signing payload
+        let ack_namespace = ack_namespace(namespace);
+        let message = Self::payload(&self.chunk, &self.epoch);
+
+        // Verify signature
+        ops::partial_verify_message(
+            identity,
+            Some(ack_namespace.as_ref()),
+            &message,
+            &self.signature,
+        )
+        .is_ok()
+    }
+
+    /// Generate a new Ack.
+    ///
+    /// This is used by validators to create and sign new acknowledgments for chunks.
+    /// It creates a partial signature over the chunk and epoch using the provided share.
+    pub fn sign(namespace: &[u8], share: &Share, chunk: Chunk<P, D>, epoch: Epoch) -> Self {
+        // Construct signing payload
+        let ack_namespace = ack_namespace(namespace);
+        let message = Self::payload(&chunk, &epoch);
+
+        // Sign message
+        let signature = ops::partial_sign_message(share, Some(ack_namespace.as_ref()), &message);
+        Self::new(chunk, epoch, signature)
+    }
+}
+
+impl<P: Array, D: Digest> Write for Ack<P, D> {
+    fn write(&self, writer: &mut impl BufMut) {
+        self.chunk.write(writer);
+        self.epoch.write(writer);
+        self.signature.write(writer);
+    }
+}
+
+impl<P: Array, D: Digest> Read for Ack<P, D> {
+    fn read_cfg(reader: &mut impl Buf, _: &()) -> Result<Self, CodecError> {
+        let chunk = Chunk::read(reader)?;
+        let epoch = Epoch::read(reader)?;
+        let signature = PartialSignature::read(reader)?;
+        Ok(Self {
+            chunk,
+            epoch,
+            signature,
+        })
+    }
+}
+
+impl<P: Array, D: Digest> FixedSize for Ack<P, D> {
+    const SIZE: usize = Chunk::<P, D>::SIZE + Epoch::SIZE + PartialSignature::SIZE;
+}
+
+/// Activity is the type associated with the [`Reporter`](crate::Reporter) trait.
+///
+/// This enum represents the two main types of activities that are reported:
+/// 1. Proposals - when a new chunk is proposed by a sequencer
+/// 2. Locks - when a threshold signature is formed for a chunk
+///
+/// The Reporter is notified of these activities so it can track the state of the system
+/// and provide the appropriate information to other components.
+#[allow(clippy::large_enum_variant)]
+#[derive(Clone, Debug, PartialEq)]
+pub enum Activity<C: Verifier, D: Digest> {
+    /// A new proposal from a sequencer
```

Comment:
Should we mention that this is only emitted if it's already verified by the application?

Reply:
Sure. Will add that!

Reply:
Added a TODO to introduce the same to `simplex` and `threshold-simplex`: https://github.com/commonwarexyz/monorepo/issues/767

Reply:
It may be better to send this before the application has verified it I suppose 🤔